### PR TITLE
refactor(*): update repo files

### PR DESCRIPTION
**Pull request by jai**
# Description

- Update .gitignore
- Remove .idea folder
- Add release workflow
- Update readme

---
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-aws-subnet/7)

### CODE REVIEW

**Comment by jai** [comment]
Blocked by #5

**Comment by bludot** [comment]
So it looks like yes it is correct as it takes the variables from inputs.tf at the root level. It however does not take in account the module required inputs.

**Review by bludot — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  -   repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v3.2.0
+      hooks:
+        -   id: trailing-whitespace
```
Comment (jai) [inline .pre-commit-config.yaml:7]: the indentation looks off/weird?

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ A simple module that creates subnets for you.
 
 ## How to use:
 
-Make sure you have a vpc and know what region you are in. 
+Make sure you have a vpc and know what region you are in.
```
Comment (jai) [inline README.md:7]: vpc -> VPC

File: .github/workflows/terraform-release-main.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        id: init
+        run: terraform init
+      - name: Terraform Validate
+        id: validate
+        run: terraform validate -no-color
+      - name: Terraform Plan
+        id: plan
+        run: terraform plan -no-color
+        continue-on-error: true
+      - name: release
```
Comment (jai) [inline .github/workflows/terraform-release-main.yml:39]: name case doesn't match other steps

File: .github/workflows/terraform-release-main.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name: "Terraform - Release"
+on:
+  push:
+    branches: [main]
+env:
+  tf_version: "latest"
+  tf_working_dir: "."
+  GOOGLE_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS_READ_ALL_PROJECTS }}
```
Comment (jai) [inline .github/workflows/terraform-release-main.yml:8]: I think this can be removed

**Review by JoseFMP — APPROVED** [review]

**Pull request merged**